### ISSUE TITLE
[ppc64le] Regular dask.distributed doesn't work on Power8 with infiniband with IBMs spectrum MPI.

### ISSUE THREAD

**Issue by ghltshubh** [dask/distributed#2341]
I am trying to install dask.distributed on IBMs power8 and NVIDIA P100 supercomputer with CentOS 7.4. It uses infiniband to communicate between the nodes. Its an LSF cluster.

**Comment by mrocklin** [comment]
Was there a particular problem that you ran into that you would like to share?

**Comment by ghltshubh** [comment]
We have a little bit different workflow

**Comment by ghltshubh** [comment]
We have a little bit different workflow
intro to Dask video uses PBS cluster and they address a process of

job submit (bsub) -> job runs on compute nodes

… but we have one more hop in the middle…

job submit (bsub) -> batch script runs on dedicated batch nodes -> job runs on compute nodes via jsrun
What do you think I should do?

**Comment by guillaumeeb** [comment]
Do you use dask-jobqueue?

Your problem seems related to what you underline in your previous comment, but we lack some information to help here. I don't think using infiniband or not is an issue here.

You should probably open a more documented issue in dask-jobqueue project (if you're using it) on the problem you are facing, with some minimalist code that reproduce your problem. See http://jobqueue.dask.org/en/latest/debug.html for some things to look at first and eventually record in your issue. Also, do you have any error output to share related to bsub command or job stdout/stderr files?

**Comment by ghltshubh** [comment]
No my problem is related to dask distributed only. I can use dask mpi but
the thing I want to know is if there is a work around to dask using mpi
only for distribution initially not interested nide communication later.

**Comment by guillaumeeb** [comment]
Did you look at [dask-jobqueue](https://github.com/dask/dask-jobqueue) which is made for spawning dask distributed cluster on HPC systems with or without Infiniband?

I don't quite get your problem, is this an issue if dask-mpi uses a more performant network interface than defaulting to ethernet?

**Comment by ghltshubh** [comment]
Yeah exactly it is about not using the performant network. I apologise for
not being clear.

**Comment by guillaumeeb** [comment]
So can you elaborate on how using high performance network is a problem on your use case?

An again, may dask-jobqueue fit your need?

**Comment by ghltshubh** [comment]
It's because the documentation says that dask distributed uses dask mpi to
distribute the jobs initially to the nodes only and not for internode
communication later.

**Comment by ghltshubh** [comment]
Ok but what constitutes ample usage. Do you mean there won't be any
performance drops?

**Comment by martindurant** [comment]
Are you experiencing poor performance with dask due to the network?

**Comment by guillaumeeb** [comment]
See #2177, there is no Comm implementation for using MPI protocol with Dask yet.

However, dask-jobqueue (and maybe dask-mpi ?) is perfectly capable of using IP over IB, which gives some improvement over standard 1Gb/s ethernet in some heavy internode exchange pattern (but as @martindurant underlines, this is probably an edge problem when using Dask).

See also http://matthewrocklin.com/blog/work/2017/09/18/pangeo-1 which has a paragraph on network performances.

And no, dask-jobqueue does not use dask-mpi, it is totally independent of MPI.

**Comment by ghltshubh** [comment]
@martindurant I haven't implemented but I am speculating based on the [documentation](http://docs.dask.org/en/latest/setup/hpc.html#using-mpi):  It only uses MPI to start the Dask cluster and not for inter-node communication.
